src/main/java/ch/njol/skript/sections/EffSecSpawn.java

```diff
@@ -142,7 +150,9 @@ protected TriggerItem walk(Event e) {
 				SpawnEvent spawnEvent = new SpawnEvent(o);
 				// Copy the local variables from the calling code to this section
 				Variables.setLocalVariables(spawnEvent, localVars);
-				trigger.execute(spawnEvent);
+				TriggerItem.walk(trigger, spawnEvent);
+				Variables.setLocalVariables(e, Variables.copyLocalVariables(spawnEvent));
+				Variables.removeLocals(spawnEvent); // Clear them since we didn't do it after Trigger execution
```

Comment:
Clarity of comment.

Also we may have to address this (not this pull request, elsewhere), but we need to be ensuring that Skript/Addon developers are removing local variables after they run their sections, as localVariables is a ConcurrentHashMap and not a WeakReference implementation collection, so the event will stay in the map.
```suggestion
				Variables.removeLocals(spawnEvent); // Clear spawnEvent local variables after execution.
```